src/pages/home/report/ReportActionItemMessageEdit.js

```diff
@@ -74,315 +74,311 @@ const defaultProps = {
+            return parser.htmlToMarkdown(draftMessage).trim();
         }
-
-        this.state = {
-            draft: draftMessage,
-            selection: {
-                start: 0,
-                end: 0,
-            },
-            isFocused: false,
-            hasExceededMaxCommentLength: false,
-        };
-    }
-
-    componentDidMount() {
+        // We need to decode saved draft message because it's escaped before saving.
+        return Str.htmlDecode(draftMessage);
+    });
+    const [selection, setSelection] = useState({start: 0, end: 0});
+    const [isFocused, setIsFocused] = useState(false);
+    const [hasExceededMaxCommentLength, setHasExceededMaxCommentLength] = useState(false);
+
+    const textInputRef = useRef(null);
+
+    // native ids
+    const saveButtonID = 'saveButton';
+    const cancelButtonID = 'cancelButton';
+    const emojiButtonID = 'emojiButton';
+    const messageEditInput = 'messageEditInput';
+
+    useEffect(() => {
         // For mobile Safari, updating the selection prop on an unfocused input will cause it to automatically gain focus
         // and subsequent programmatic focus shifts (e.g., modal focus trap) to show the blue frame (:focus-visible style),
         // so we need to ensure that it is only updated after focus.
-        this.setState((prevState) => ({
-            selection: {
-                start: prevState.draft.length,
-                end: prevState.draft.length,
-            },
-        }));
-    }
-
-    componentWillUnmount() {
-        // Skip if this is not the focused message so the other edit composer stays focused.
-        if (!this.state.isFocused) {
-            return;
-        }
+        setDraft((prevDraft) => {
+            setSelection({
+                start: prevDraft.length,
+                end: prevDraft.length,
+            });
+            return prevDraft;
+        });
+    }, []);
 
-        // Show the main composer when the focused message is deleted from another client
-        // to prevent the main composer stays hidden until we swtich to another chat.
-        ComposerActions.setShouldShowComposeInput(true);
-    }
+    useEffect(
+        () => () => {
+            // Skip if this is not the focused message so the other edit composer stays focused.
+            if (!isFocused) {
+                return;
+            }
+
+            // Show the main composer when the focused message is deleted from another client
+            // to prevent the main composer stays hidden until we swtich to another chat.
+            ComposerActions.setShouldShowComposeInput(true);
+        },
+        // eslint-disable-next-line react-hooks/exhaustive-deps -- for willUnmount lifecycle
+        [],
+    );
 
     /**
      * Update Selection on change cursor position.
      *
      * @param {Event} e
      */
-    onSelectionChange(e) {
-        this.setState({selection: e.nativeEvent.selection});
-    }
+    const onSelectionChange = useCallback((e) => {
```

Comment:
I think we should just inline this function since it's used only once and there's almost no cost to declaring it.

Reply:
NAB though because imo our [guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#should-i-wrap-all-my-inline-functions-with-usecallback-or-move-them-out-of-the-component-if-they-have-no-dependencies) here aren't very clear.